src/utils/form-actions.ts

```diff
@@ -2,108 +2,56 @@
+  return currentTimeSeconds + maxTTLDurationInSeconds;
 };
 
-export async function createSession(session: Omit<Session, 'id'>) {
-  const sessionWithTTL = {
-    ...session,
+export async function createTemplate(template: Omit<Template, 'id'>) {
+  const templateWithTTL = {
+    ...template,
     ttl: calculateTTL(),
   };
   const { data, errors } =
-    await getAmplifyBackendClient().models.SessionStorage.create(
-      sessionWithTTL
+    await getAmplifyBackendClient().models.TemplateStorage.create(
+      templateWithTTL
     );
 
   if (errors) {
-    logger.error('Failed to create session', errors);
+    logger.error('Failed to create template', errors);
     throw new Error('Failed to create new template');
   }
   return data;
 }
 
-export async function saveSession(session: Session) {
+export async function saveTemplate(template: Template, ttl = false) {
   const { data, errors } =
-    await getAmplifyBackendClient().models.SessionStorage.update({
-      ...session,
-      ttl: calculateTTL(),
+    await getAmplifyBackendClient().models.TemplateStorage.update({
+      ...template,
+      ...(ttl && { ttl: calculateTTL() }),
     });
 
   if (errors) {
-    logger.error('Failed to save session', errors);
+    logger.error('Failed to save template', errors);
     throw new Error('Failed to save template data');
   }
+
   if (!data) {
     throw new DbOperationError({
       message:
-        'Session in unknown state. No errors reported but entity returned as falsy',
+        'Template in unknown state. No errors reported but entity returned as falsy',
       operation: 'update',
     });
   }
-  return data as Session;
-}
-
-export async function getSession(
-  sessionId: string
-): Promise<Session | undefined> {
-  const { data, errors } =
-    await getAmplifyBackendClient().models.SessionStorage.get({
-      id: sessionId,
-    });
-  if (errors) {
-    logger.error('Failed to get session', errors);
-  }
-
-  if (!data) {
-    logger.warn(`Failed to retrieve session for ID ${sessionId}`);
-    return undefined;
-  }
-
-  return data as Session;
-}
-
-export async function saveTemplate(template: Omit<Template, 'id'>) {
-  const backendClient = getAmplifyBackendClient();
-  const templateRepository = backendClient.models.TemplateStorage;
-
-  const { data, errors } = await templateRepository.create({
-    ...template,
-    id: `${template.type}-${randomUUID()}`.toLowerCase(),
-  });
-
-  if (errors) {
-    throw new DbOperationError({
-      message: `Failed saving ${template.type} template`,
-      operation: 'create',
-      cause: errors,
-    });
-  }
-
-  if (!data) {
-    throw new DbOperationError({
-      message: `${template.type} template entity in unknown state. No errors reported but entity returned as falsy`,
-      operation: 'create',
-      cause: [
-        {
-          message: 'Fields attempting to be saved',
-          data: template,
-        },
-      ],
-    });
-  }
   return data as Template;
```

Comment:
I wonder now that we have updated typings whether doing `return data satisfies Template` instead of casting the response to Template.

Reply:
sadly not - Amplify data does not support marking enum fields as required so without the typecast you get a problem where Amplify says templateType can be null, but our code always guarantees that it exists

Reply:
God Amplify is a pain! Thanks for trying!